### ISSUE TITLE
Slowness makes you fall through the earth and you can't move after affected by it

### ISSUE THREAD

**Issue by brandyyn** [makamys/SmartMoving#19]
Slowness makes you fall through the earth and you can't move after affected by it

Steps to recreate:
Get hit by an Advent of Ascension Sea Spider, it will affect you with "Slowness CI" and you fall through the earth, this also happens with the Stun potion effect from skeletons when using TragicMC2

You can't interact with the world at all afterwards.

**Comment by brandyyn** [comment]
or use the /effect name 2 7 7 and jump and you'll fall straight through the earth :)

**Comment by DarianLStephens** [comment]
I wonder if it's the same sort of problem that BetterFPS caused with Et Futurum's Elytra? Bad equations resulting in NaN coordinates?

**Comment by Darek505** [comment]
Yes, that's true. 

This happens at level 6 slowness , when the slow coefficient is (0.15 * (lvl+1)) = 1.05. In vanilla, (since the slowdown factor is > 1), speed = 1.0 - 1.05 = -0.05 (< 0, so actually 0) the player stops completely. But in smart moving, there are some parts that use division by speed, which causes n/0 = Infinity. Which ultimately causes the player's speed to become NaN, which means the coordinates are also set to NaN

**Comment by brandyyn** [comment]
> Yes, that's true.
> 
> This happens at level 6 slowness , when the slow coefficient is (0.15 * (lvl+1)) = 1.05. In vanilla, (since the slowdown factor is > 1), speed = 1.0 - 1.05 = -0.05 (< 0, so actually 0) the player stops completely. But in smart moving, there are some parts that use division by speed, which causes n/0 = Infinity. Which ultimately causes the player's speed to become NaN, which means the coordinates are also set to NaN

Yes exactly what happens to me, anyway hopefully to help with the bug the Alfheim devs did this for me: 
![IMG_5327](https://github.com/user-attachments/assets/0b243571-e25f-416e-99d7-df7f92f8de21)

**Comment by brandyyn** [comment]
![image](https://github.com/user-attachments/assets/7ea25683-bf62-458b-b8c6-9d10d02b97d2)

**Comment by brandyyn** [comment]
potion effects start from 0, so level 0 = level 1 potion effect
![image](https://github.com/user-attachments/assets/41b2245e-524e-4ae5-9b15-23238578a33f)

**Comment by brandyyn** [comment]
Fixed by https://github.com/brandyyn/SmartMoving/commit/a9984d1fcf51ce19bd907e773f745b5fcd0498fe
